storage/remote/azuread/azuread.go

```diff
@@ -46,11 +47,26 @@ type ManagedIdentityConfig struct {
 	ClientID string `yaml:"client_id,omitempty"`
 }
 
+// OAuthConfig is used to store azure oauth config values.
+type OAuthConfig struct {
+	// ClientID is the clientId of the azure active directory application that is being used to authenticate.
+	ClientID string `yaml:"client_id,omitempty"`
+
+	// ClientSecret is the clientSecret of the azure active directory application that is being used to authenticate.
+	ClientSecret string `yaml:"client_secret,omitempty"`
+
+	// TenantID is the tenantId of the azure active directory application that is being used to authenticate.
+	TenantID string `yaml:"tenant_id,omitempty"`
+}
+
 // AzureADConfig is used to store the config values.
 type AzureADConfig struct { // nolint:revive
 	// ManagedIdentity is the managed identity that is being used to authenticate.
 	ManagedIdentity *ManagedIdentityConfig `yaml:"managed_identity,omitempty"`
 
+	// OAuth is the oauth config that is being used to authenicate.
+	OAuth *OAuthConfig `yaml:"oauth,omitempty"`
+
 	// Cloud is the Azure cloud in which the service is running. Example: AzurePublic/AzureGovernment/AzureChina.
 	Cloud string `yaml:"cloud,omitempty"`
 }
```

Comment:
How about this? Put only what is in oauth in OAuthConfig and put only in azure in AzureADConfig

```suggestion
// OAuthConfig is used to store azure oauth config values.
type OAuthConfig struct {
	// ClientID is the clientId of the azure active directory application that is being used to authenticate.
	ClientID string `yaml:"client_id,omitempty"`

	// ClientSecret is the clientSecret of the azure active directory application that is being used to authenticate.
	ClientSecret string `yaml:"client_secret,omitempty"`
}

// AzureADConfig is used to store the config values.
type AzureADConfig struct { // nolint:revive
	// ManagedIdentity is the managed identity that is being used to authenticate.
	ManagedIdentity *ManagedIdentityConfig `yaml:"managed_identity,omitempty"`

	// OAuth is the oauth config that is being used to authenicate.
	OAuth *OAuthConfig `yaml:"oauth,omitempty"`

	// Cloud is the Azure cloud in which the service is running. Example: AzurePublic/AzureGovernment/AzureChina.
	Cloud string `yaml:"cloud,omitempty"`

	// TenantID is the tenantId of the azure active directory application that is being used to authenticate.
	TenantID string `yaml:"tenant_id,omitempty"`
}
```

testdata example
```
cloud: AzurePublic
tenant_id: 00000000-a12b-3cd4-e56f-000000000000
oauth:
  client_id: 00000000-0000-0000-0000-000000000000
  client_secret: Cl1ent$ecret!
```